plugin/azure/azure_test.go

```diff
@@ -0,0 +1,149 @@
+package azure
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/coredns/coredns/plugin/file"
+	"github.com/coredns/coredns/plugin/pkg/dnstest"
+	"github.com/coredns/coredns/plugin/pkg/fall"
+	"github.com/coredns/coredns/plugin/test"
+	crequest "github.com/coredns/coredns/request"
+	"github.com/miekg/dns"
+)
+
+var demoAzure Azure = Azure{
+	Next:      returnTestHandler(),
+	Fall:      fall.Zero,
+	zoneNames: []string{"example.org", "www.example.org", "example.org.", "sample.example.org."},
+	zones:     returnTestZones(),
+}
+
+func returnTestZones() zones {
+	zones := make(map[string][]*zone)
+	zones["example.org."] = append(zones["example.org."], &zone{dns: "example.org."})
+	newZ := file.NewZone("example.org.", "")
+
+	for _, rr := range []string{
+		"example.org.  300 IN  A   1.2.3.4",
+		"example.org.  300 IN  AAAA   2001:db8:85a3::8a2e:370:7334",
+		"www.example.org.  300 IN  A   1.2.3.4",
+		"www.example.org.  300 IN  A   1.2.3.4",
+		"org.	172800	IN	NS	ns3-06.azure-dns.org.",
+		"org.	300	IN	SOA	ns1-06.azure-dns.com. azuredns-hostmaster.microsoft.com. 1 3600 300 2419200 300",
+	} {
+		r, _ := dns.NewRR(rr)
+		err := newZ.Insert(r)
+		if err != nil {
+			log.Fatalf("XXXXXXXXXXXXXX could not add file zone to zone: %s", err.Error())
+		}
+	}
+	zones["example.org."][0].z = newZ
+	return zones
+}
+
+func returnTestHandler() test.HandlerFunc {
+	return func(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+		state := crequest.Request{W: w, Req: r}
+		qname := state.Name()
+		m := new(dns.Msg)
+		rcode := dns.RcodeServerFailure
+		if qname == "example.gov." {
```

Comment:
example.org

Reply:
this is to test for fallthrough, it should be `example.gov` no?

Reply:
ok, that warrents a comment then